opencti-platform/opencti-graphql/src/modules/xtm/xtm-domain.js

```diff
@@ -327,67 +340,71 @@ export const generateOpenBasScenario = async (context, user, stixCoreObject, att
       );
       dependsOnDuration += (interval * 60);
     } else {
-      const obasInjectorContracts = await getInjectorContracts(obasAttackPattern.attack_pattern_id);
-      let finalObasInjectorContracts = R.take(5, getShuffledArr(obasInjectorContracts));
-      if (selection === 'random') {
-        finalObasInjectorContracts = R.take(1, finalObasInjectorContracts);
-      }
-      if (simulationType === 'technical') {
-        // eslint-disable-next-line no-restricted-syntax
-        for (const finalObasInjectorContract of finalObasInjectorContracts) {
-          const obasInjectorContractContent = JSON.parse(finalObasInjectorContract.injector_contract_content);
-          const title = `[${obasAttackPattern.attack_pattern_external_id}] ${obasAttackPattern.attack_pattern_name} - ${finalObasInjectorContract.injector_contract_labels.en}`;
-          await createInjectInScenario(
-            obasScenario.scenario_id,
-            obasInjectorContractContent.config.type,
-            finalObasInjectorContract.injector_contract_id,
-            title,
-            dependsOnDuration,
-            null,
-            [{ value: 'opencti', color: '#001bda' }, { value: 'technical', color: '#b9461a' }]
-          );
-          dependsOnDuration += (interval * 60);
-        }
+      const obasInjectorContracts = await searchInjectorContracts(obasAttackPattern.attack_pattern_external_id, platforms, architecture);
+
+      if (obasInjectorContracts.length === 0) {
+        attackPatternsWithoutInjectorContracts.push(obasAttackPattern.attack_pattern_external_id);
+        logApp.info(`[GENERATION SCENARIO OBAS] No injector contracts available for this attack pattern ${obasAttackPattern.attack_pattern_external_id}`);
       } else {
-        // TODO
+        let finalObasInjectorContracts = R.take(5, getShuffledArr(obasInjectorContracts));
```

Comment:
try not tu use Ramda when it's possible
R.take(number, array) —> array.slice(0, number)